webapp/channels/src/components/widgets/modals/components/section_creator.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+
+import './section_creator.scss';
+
+type Props = {
+    title: {
+        id: string;
+        defaultMessage: string;
+    };
+    description?: {
+        id: string;
+        defaultMessage: string;
+    };
+    content: JSX.Element;
+    titleSuffix?: JSX.Element;
+};
+
+function SectionCreator({
+    title,
+    description,
+    content,
+    titleSuffix,
+}: Props): JSX.Element {
+    const {formatMessage} = useIntl();
+    const Title = (
```

Comment:
Nit, instead of `Title` and `Description` could you rename these to something like `titleContent` or `titleComponent`? The capitalized name makes me think they're classes or React components currently